### PR TITLE
Add share identifier to aws batch

### LINKED ISSUE #3088: Add support for AWS Batch share identifier
AWS Batch queue now supports advanced scheduling policy that allows users to differentiate their jobs from one another via a *share identifier*. See [here for details](https://aws.amazon.com/blogs/hpc/introducing-fair-share-scheduling-for-aws-batch/).  

The goal of this issue is to allow Nextflow users to specify a value for the share identifier to be used by all the tasks submitted by the nextflow workflow. 

The attribute should be set via the following setting in the nextflow config file 

```
aws.batch.shareIdentifier 
```

This value should be added in the job requests submitted by nextflow

### PR DESCRIPTION
**Pull request by bentsherman**
Closes #3088

### CODE REVIEW

File: docs/config.rst [hunk 1]
```diff
@@ -171,6 +171,7 @@ delayBetweenAttempts        Delay between download attempts from S3 (default `10
 maxParallelTransfers        Max parallel upload/download transfer operations *per job* (default: ``4``).
 maxTransferAttempts         Max number of downloads attempts from S3 (default: `1`).
 maxSpotAttempts             Max number of execution attempts of a job interrupted by a EC2 spot reclaim event (default: ``5``, requires ``22.04.0`` or later)
+shareIdentifier             The share identifier for all tasks when using `fair-share scheduling for AWS Batch <https://aws.amazon.com/blogs/hpc/introducing-fair-share-scheduling-for-aws-batch/>`_
```
Comment (pditommaso) [inline docs/config.rst:174]: Let's mention 

```
 (requires ``22.09.0-edge`` or later)
```

**Comment by pditommaso** [comment]
@bentsherman You cancelled the build?

**Comment by bentsherman** [comment]
Forgot to add a [ci skip] to the docs commit. Trying to save some trees. Looks like the first commit passed

**Comment by pditommaso** [comment]
> Trying to save some trees

That's good! Just burning a small bush to be sure everything is fine 😄

**Comment by pditommaso** [comment]
All fine. Let's merge it!

**Pull request merged**